src/components/MAPI/clusters/CreateCluster/index.tsx

```diff
@@ -0,0 +1,330 @@
+
+import {
+  createCluster,
+  createDefaultCluster,
+  createDefaultControlPlaneNode,
+  createDefaultProviderCluster,
+  findLatestReleaseVersion,
+} from '../utils';
+import WorkerNodesCreateClusterDescription from './CreateClusterDescription';
+import CreateClusterName from './CreateClusterName';
+import {
+  ClusterPatch,
+  IClusterPropertyValue,
+  withClusterReleaseVersion,
+} from './patches';
+
+enum ClusterPropertyField {
+  Name,
+  Description,
+}
+
+interface IApplyPatchAction {
+  type: 'applyPatch';
+  property: ClusterPropertyField;
+  value: ClusterPatch;
+}
+
+interface IChangeValidationResultAction {
+  type: 'changeValidationStatus';
+  property: ClusterPropertyField;
+  value: boolean;
+}
+
+interface IStartCreationAction {
+  type: 'startCreation';
+}
+
+interface IEndCreationAction {
+  type: 'endCreation';
+}
+
+interface ISetLatestReleaseAction {
+  type: 'setLatestRelease';
+  value: string;
+}
+
+type ClusterAction =
+  | IApplyPatchAction
+  | IChangeValidationResultAction
+  | IStartCreationAction
+  | IEndCreationAction
+  | ISetLatestReleaseAction;
+
+interface IClusterState {
+  provider: PropertiesOf<typeof Providers>;
+  cluster: Cluster;
+  providerCluster: ProviderCluster;
+  controlPlaneNode: ControlPlaneNode;
+  validationResults: Record<ClusterPropertyField, boolean>;
+  isCreating: boolean;
+  latestRelease: string;
+}
+
+interface IReducerConfig {
+  namespace: string;
+  organization: string;
+}
+
+function makeInitialState(
+  provider: PropertiesOf<typeof Providers>,
+  config: IReducerConfig
+): IClusterState {
+  const name = generateUID(5);
+
+  const resourceConfig = { ...config, name, releaseVersion: '' };
+  const providerCluster = createDefaultProviderCluster(
+    provider,
+    resourceConfig
+  );
+  const controlPlaneNode = createDefaultControlPlaneNode({ providerCluster });
+  const cluster = createDefaultCluster({ providerCluster });
+
+  return {
+    provider,
+    cluster,
+    providerCluster,
+    controlPlaneNode,
+    validationResults: {
+      [ClusterPropertyField.Name]: true,
+      [ClusterPropertyField.Description]: true,
+    },
+    isCreating: false,
+    latestRelease: '',
+  };
+}
+
+const reducer: React.Reducer<IClusterState, ClusterAction> = produce(
+  (draft, action) => {
+    switch (action.type) {
+      case 'applyPatch':
+        action.value(
+          draft.cluster,
+          draft.providerCluster,
+          draft.controlPlaneNode
+        );
+        break;
+      case 'changeValidationStatus':
+        draft.validationResults[action.property] = action.value;
+        break;
+      case 'startCreation':
+        draft.isCreating = true;
+        break;
+      case 'endCreation':
+        draft.isCreating = false;
+        break;
+      case 'setLatestRelease':
+        // Apply the version to the CRs if no version was set before.
+        if (!draft.latestRelease) {
+          withClusterReleaseVersion(action.value)(
```

Comment:
Since we need to fetch all the releases first in order to determine the latest release version, once we have all the data necessary we update the resources.